### PR TITLE
Prefix old config provider types with `Obsolete`

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Part of #4168

### CODE REVIEW

File: src/Integration.UnitTests/NewConnectedMode/ObsoleteConfigurationProviderTests.cs [hunk 1]
```diff
@@ -34,12 +31,12 @@
 namespace SonarLint.VisualStudio.Integration.UnitTests
 {
     [TestClass]
-    public class ConfigurationProviderTests
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/NewConnectedMode/ObsoleteConfigurationProviderTests.cs:37]: No code changes - I just renamed the existing types and files.

File: src/Core/Binding/IConfigurationProvider.cs [hunk 2]
```diff
@@ -22,6 +22,9 @@ namespace SonarLint.VisualStudio.Core.Binding
 {
     public interface IConfigurationProvider
     {
+        /// <summary>
+        /// Returns the binding configuration for the current solution
+        /// </summary>
```
Comment (duncanp-sonar) [inline src/Core/Binding/IConfigurationProvider.cs:27]: Miscellaneous - added a doc comment to the interface

File: src/Integration/MefServices/VsSessionHost.cs [hunk 3]
```diff
@@ -298,7 +299,8 @@ private void RegisterLocalServices()
         {
             this.localServices.Add(typeof(ICredentialStoreService), new Lazy<ILocalService>(() => credentialStoreService));
 
-            this.localServices.Add(typeof(IConfigurationProviderService), new Lazy<ILocalService>(() => new ConfigurationProvider(this, credentialStoreService, Logger)));
+            this.localServices.Add(typeof(IObsoleteConfigurationProviderService), new Lazy<ILocalService>(() => new ObsoleteConfigurationProvider(this, credentialStoreService, Logger)));
+            this.localServices.Add(typeof(IConfigurationProviderService), new Lazy<ILocalService>(() => new UnintrusiveConfigurationProvider()));
```
Comment (duncanp-sonar) [inline src/Integration/MefServices/VsSessionHost.cs:303]: The config provider is obtained via a local service provide i.e. the service needs to be registered and created (we'll get rid of this mechanism at some point in the future).

Both the old and new service implementations are registered.

File: src/Integration/UnintrusiveBinding/UnintrusiveConfigurationProvider.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    internal interface IConfigurationProviderService : IConfigurationProvider, ILocalService
+    {
+    }
+
+    [Export(typeof(IConfigurationProvider))]
+    [PartCreationPolicy(CreationPolicy.Any)]
+    internal class UnintrusiveConfigurationProvider : IConfigurationProviderService
+    {
+        // TODO - fetch settings from new location #4170
+        public BindingConfiguration GetConfiguration() => BindingConfiguration.Standalone;
```
Comment (duncanp-sonar) [inline src/Integration/UnintrusiveBinding/UnintrusiveConfigurationProvider.cs:35]: Place holder for the new service.

**Review by bigfluffycookie — APPROVED** [review]
Looks good!

**Pull request merged**